Source/items.cpp

```diff
@@ -4575,8 +4566,6 @@ void SpawnWitch(int lvl)
 
 		item._iCreateInfo = lvl | CF_WITCH;
 		item._iIdentified = true;
-		WitchBookLevel(item);
-		item._iStatFlag = StoreStatOk(item);

```

Comment:
Moving this to when the vendor is accessed makes the code more generic and also doesn't require part of it to be hacked in to unpacking of the player struct.

Reply:
I believe this is done already... ?

Reply:
Correct i extracted this part in to its own PR which has been merged